modules/index/src/main/java/org/apache/ignite/internal/index/IndexManager.java

```diff
@@ -410,21 +394,18 @@ private void startIndexes() {
         int catalogVersion = catalogManager.latestCatalogVersion();
         long causalityToken = recoveryFinishedFuture.join();
 
-        NamedListView<TableView> tableListView = tablesCfg.tables().value();
-
         List<CompletableFuture<?>> startIndexFutures = new ArrayList<>();
 
         for (CatalogIndexDescriptor index : catalogManager.indexes(catalogVersion)) {
-            CatalogTableDescriptor table = catalogManager.table(index.tableId(), catalogVersion);
+            int tableId = index.tableId();
 
-            assert table != null : "tableId=" + index.tableId() + ", indexId=" + index.id();
+            CatalogTableDescriptor table = catalogManager.table(tableId, catalogVersion);
 
-            // TODO: IGNITE-19499 Only catalog should be used
-            int configTableId = getConfigTableId(table.name(), tableListView);
+            assert table != null : "tableId=" + tableId + ", indexId=" + index.id();
 
-            CompletableFuture<?> fireCreateIndexEventFuture = fireCreateIndexEvent(index, causalityToken, configTableId);
+            CompletableFuture<?> fireCreateIndexEventFuture = fireCreateIndexEvent(index, causalityToken, tableId);
```

Comment:
1. Do we need IndexEvents? Seems, no one use them.
2. If yes, then will it be correctly to register index in table, then fire the event?

Reply:
Maybe we should get rid of index events altogether, in the sense of deleting them to hell?

Reply:
I looked at the opportunity to get rid of this, I realized that I need to make quite a few changes for this, I propose to do it in IGNITE-20339.